core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -926,7 +926,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // The call to new NewHadoopJob automatically adds security credentials to conf,
     // so we don't need to explicitly add them ourselves
     val job = new NewHadoopJob(conf)
-    NewFileInputFormat.addInputPath(job, new Path(path))
+    // Use addInputPaths so that newAPIHadoopFile aligns with hadoopFile in taking
+    // comma separated files as input. (see SPARK-7155)
+    NewFileInputFormat.addInputPaths(job, path)
```

Comment:
I don't think that was the intent of the API -- you specify the paths as an argument, and it could be surprising to also include something that happens to be in the existing config already.